system-x/services/knative/src/main/java/software/tnb/knative/validation/KnativeValidation.java

```diff
@@ -55,13 +55,18 @@ public InMemoryChannel createInMemoryChannel(String name) {
 
     public Broker createBroker(String name) {
         validateName(name);
+        List<Broker> brokerList =
+            client.brokers().list().getItems().stream().filter(broker -> broker.getMetadata().getName().equals(name)).collect(Collectors.toList());
```

Comment:
`if (client.brokers().withName(name).get() != null)`

Reply:
I tried this one too, however adding this line caused it strangely ended in same problem (I haven't investigated it properly)

Reply:
didn't you miss the `.get()` at the end? because `client.brokers().withName(name)` is always not null

Reply:
Nn, I debugged it in Idea via `Evaluate expression` - ie. I knew what was inside in moment...

Reply:
My first attempt was
```
public Broker createBroker(String name) {
        validateName(name);
        if (client.brokers().withName(name).get() != null) {
            return client.brokers().withName(name).get();
        }
        final Broker broker = new BrokerBuilder()
            .withNewMetadata()
            .withName(name)
            .endMetadata()
            .build();
        client.brokers().create(broker);
        createdItems.add(broker);
        return broker;
}
```
but then the broker was again not Ready (original problem)...

Reply:
ok, it seems the validation regex changed from the time I was writing this class :grinning: the problem with the broker in your case is that the name can't start with a digit. If you create a broker `9cd2d` the related event (`oc get events`) is:

```
Service "9cd2d-kne-trigger-kn-channel" is invalid: metadata.name: Invalid value: "9cd2d-kne-trigger-kn-channel": a DNS-1035 label must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character (e.g. 'my-name',  or 'abc-123', regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?')
```

Similarly, when you create an in-memory channel with name `9cd2d`, the event is:

```
Warning   InternalError   inmemorychannel/9cd2d               Service "9cd2d-kn-channel" is invalid: metadata.name: Invalid value: "9cd2d-kn-channel": a DNS-1035 label must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character (e.g. 'my-name',  or 'abc-123', regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?')
```